target/linux/ramips/dts/mt7621_dlink_dir-xx60-a1.dtsi

```diff
@@ -0,0 +1,167 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+
+#include "mt7621.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	aliases {
+		label-mac-device = &gmac0;
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "reset";
+			gpios = <&gpio 15 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+
+		wps {
+			label = "wps";
+			gpios = <&gpio 18 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_WPS_BUTTON>;
+		};
+	};
+};
+
+&nand {
+	status = "okay";
+
+	partitions {
+		compatible = "fixed-partitions";
+		#address-cells = <1>;
+		#size-cells = <1>;
+
+		partition@0 {
+			label = "Bootloader";
+			reg = <0x0 0x80000>;
+			read-only;
+		};
+
+		partition@80000 {
+			label = "config";
+			reg = <0x80000 0x80000>;
+			read-only;
+		};
+
+		factory: partition@100000 {
+			label = "factory";
+			reg = <0x100000 0x40000>;
+			read-only;
+		};
+
+		partition@140000 {
+			label = "config2";
+			reg = <0x140000 0x40000>;
+			read-only;
+		};
+		
+		partition@180000 {
+			label = "firmware";
+			compatible = "sge,uimage";
+			reg = <0x180000 0x2800000>;
+		};
+
+		partition@2980000 {
+			label = "private";
+			reg = <0x2980000 0x2000000>;
+			read-only;
+		};
+
+		partition@4980000 {
+			label = "firmware2";
+			compatible = "sge,uimage";
+			reg = <0x4980000 0x2800000>;
+		};
+
+		partition@7180000 {
+			label = "mydlink";
+			reg = <0x7180000 0x600000>;
+			read-only;
+		};
+
+		partition@7780000 {
+			label = "reserved";
+			reg = <0x7780000 0x880000>;
+			read-only;
+		};
+	};
+};
+
+&pcie {
+	status = "okay";
+};
+
+&pcie0 {
+	wifi@0,0 {
+		compatible = "mediatek,mt76";
+		reg = <0x0000 0 0 0 0>;
+		mediatek,mtd-eeprom = <&factory 0x0000>;
+		ieee80211-freq-limit = <2400000 2500000>;
+		mtd-mac-address = <&factory 0xe000>;
+		mtd-mac-address-increment = <1>;
+
+		led {
+			led-active-low;
+		};
+	};
+};
+
+&pcie1 {
+	wifi@0,0 {
+		compatible = "mediatek,mt76";
+		reg = <0x0000 0 0 0 0>;
+		mediatek,mtd-eeprom = <&factory 0x8000>;
+		ieee80211-freq-limit = <5000000 6000000>;
+		mtd-mac-address = <&factory 0xe000>;
+		mtd-mac-address-increment = <2>;
```

Comment:
Are there no addresses provided by caldata, i.e. in 0x4 and 0x8004?

Reply:
Not on the DIR-1960 just random numbers that don't match the one's used int the OEM firmware "00223344AA00" & "000C12006028" in mine 
should be "60 63 4C 66 E4 EC" & "60 63 4C 66 E4 ED"

Reply:
So, you have addresses in 0xe000 and 0xe006 but not in 0x4 and 0x8004? That's interesting, quite uncommon.

In that case, I'd be interested in an overview of how addresses _should_ be on that device:
https://openwrt.org/docs/guide-developer/device-support-policies#documentation

That could be added to the commit message for that device then.

Same should be done for the 2660 as well.

Reply:
DIR-1960
lan          factory 0xe000   *:EB (label)
wan        factory 0xe006   *:EE
2.4 GHz  factory 0xe000+1 *:EC
5.0 GHz  factory 0xe000+2 *:ED
seems like they didn't replace the dummy entry's in the calibration data
I was expecting the same as the DIR-878 but it was not

Reply:
Yes I'll have a look.

Reply:
just need to look at you MTD2 file in a hex editor at stated locations
I'm happy to look if you can get me your MTD2 file